packages/loopring_v3/contracts/impl/libexchange/ExchangeGenesis.sol

```diff
@@ -44,8 +45,10 @@ library ExchangeGenesis
         S.genesisMerkleRoot = _genesisMerkleRoot;
         S.DOMAIN_SEPARATOR = _domainSeperator;
 
-        ILoopringV3 loopring = ILoopringV3(_loopringAddress);
+        ILoopringV3 loopring = ILoopringV3(_loopring);
         S.loopring = loopring;
+
+        S.agentRegistry = IAgentRegistry(loopring.agentRegistryAddress());
```

Comment:
Saw this more of a per exchange kind of thing, but everything will probably happen inside a single exchange contract any way so probably doesn't matter. :)

Reply:
Makes a lot of sense. Code changed accordingly.